### PR TITLE
feat: add network interface statistics collector

### PR DESCRIPTION
**Pull request by jra3**
## Summary
- Implements network interface statistics collection from `/proc/net/dev` and `/sys/class/net`
- Provides comprehensive network metrics including bytes, packets, errors, and interface metadata
- Follows established collector patterns with robust error handling and graceful degradation

## Changes
- Added `NetworkCollector` in `pkg/performance/collectors/network.go` that:
  - Collects receive/transmit statistics for all network interfaces
  - Reads interface metadata (speed, duplex, operational state, link detection) from sysfs
  - Handles virtual interfaces gracefully when sysfs metadata is unavailable
  - Implements proper error handling: critical `/proc/net/dev` failures return errors, optional sysfs failures are logged
- Added comprehensive test coverage in `pkg/performance/collectors/network_test.go`
- Uses existing `NetworkStats` type from `pkg/performance/types.go`

## Technical Details
The collector follows all established patterns:
- ✅ Constructor with absolute path validation
- ✅ Compile-time interface check
- ✅ Clear error handling strategy (critical vs optional files)
- ✅ External test package (`collectors_test`)
- ✅ Comprehensive test coverage including edge cases
- ✅ Well-documented with kernel references

### Data Sources
- **`/proc/net/dev`**: Primary source for interface statistics
  - RX: bytes, packets, errors, dropped, FIFO, frame, compressed, multicast
  - TX: bytes, packets, errors, dropped, FIFO, collisions, carrier, compressed
- **`/sys/class/net/[interface]/`**: Optional metadata
  - `speed`: Link speed in Mbps
  - `duplex`: Full or half duplex
  - `operstate`: Operational state (up/down/unknown)
  - `carrier`: Physical link detection (0/1)

## Test Plan
- [x] Unit tests pass with mock /proc and /sys filesystems
- [x] Constructor validation tests
- [x] Malformed data handling tests
- [x] Graceful degradation when sysfs unavailable
- [x] Boundary condition tests (zero values, max uint64)
- [x] Build succeeds (`make build`)
- [x] Linting passes (`make lint`)
- [x] License headers present (`make gen-license-headers`)
- [ ] Manual testing on systems with various interface types (physical, virtual, wireless)
- [ ] Verify collector works in containerized environments with custom proc/sys paths

## Notes
- Currently implements one-shot collection only
- Future enhancement could add continuous collection with rate calculations (bytes/sec, packets/sec)
- Virtual interfaces (lo, docker0, etc.) are handled gracefully when they lack sysfs metadata

### CODE REVIEW

**Comment by jra3** [comment]
@claude review

File: pkg/performance/collectors/network.go [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/antimetal/agent/pkg/performance"
+	"github.com/go-logr/logr"
+)
+
```
Comment (haq204) [inline pkg/performance/collectors/network.go:21]: missing collector registration

**Review by haq204 — CHANGES_REQUESTED** [review]
